core/src/main/java/nl/nn/adapterframework/jms/MessagingSource.java

```diff
@@ -153,7 +154,13 @@ public Object getManagedConnectionFactory() {
 		ConnectionFactory qcf = null;
 		try {
 			qcf = getConnectionFactoryDelegate();
+			if (qcf instanceof PoolingConnectionFactory) { //BTM
+				return ((PoolingConnectionFactory)qcf).getXaConnectionFactory();
+			}
 			try {
+				if (qcf instanceof ConnectionFactoryProxy) { // Narayana
+					return ClassUtils.getDeclaredFieldValue(qcf, ConnectionFactoryProxy.class, "xaConnectionFactory");
+				}
```

Comment:
Moet dit niet gewrapped worden in een `TransactionAwareConnectionFactoryProxy` ?

Reply:
Nee, dit wordt alleen maar gebruikt om beschrijvende informatie op te halen